Distributions.ecl

```diff
@@ -0,0 +1,407 @@
+    REAL8 B55 := -1920.0;
+    REAL8 B56 := -945.0;
+    REAL8 Norm_PPF := Normal_PPF(x);
+    // df 7 and higher
+    d1 := Norm_PPF;
+    d3 := d1 * d1 * d1;
+    d5 := d3 * d1 * d1;
+    d7 := d5 * d1 * d1;
+    d9 := d7 * d1 * d1;
+    df_2 := df * df;
+    df_3 := df_2 * df;
+    df_4 := df_3 * df;
+    trm1 := d1;
+    trm2 := B21*(d3+d1)/df;
+    trm3 := B31*(B32*d5+B33*d3+B34*d1)/df_2;
+    trm4 := B41*(B42*d7+B43*d5+B44*d3+B45*d1)/df_3;
+    trm5 := B51*(B52*d9+B53*d7+B54*d5+B55*d3+B56*d1)/df_4;
+    trms := trm1 + trm2 + trm3 + trm4 + trm5;
+    // df 3, 4, 5, and 6 cases
+    init_z := ATAN(Norm_PPF/SQRT(df));
+    // df 7 to 50
+    INTEGER4 ndf := (INTEGER4) df;
+    BOOLEAN isEven := ndf&1 = 0;
+    REAL8 m_factor := Math.DoubleFac(ndf-1)
+                    /(IF(isEven,2,PI)*SQRT(ndf)*Math.DoubleFac(ndf-2));
+    //
+    rslt := MAP(df=1    => -COS(PI*x)/SIN(PI*x),
+                df=2    => ((ROOT_2/2.0)*(2.0*x-1.0))/SQRT(x*(1.0-x)),
+                df=3    => df3_helper((PI*(x-0.5)), init_z),
+                df=4    => df4_helper((2.0*(x-0.5)), init_z),
+                df=5    => df5_helper((PI*(x-0.5)), init_z),
+                df=6    => df6_helper((2.0*(x-0.5)), init_z),
+                df BETWEEN 7 AND 50 => dk_helper(x, ndf, m_factor),
+                trms);
+    RETURN rslt;
+  END;
+
+  // helper functions for Chi Squared
+  REAL8 Low_DF_Sum(INTEGER4 df, REAL8 x) := BEGINC++
+    #option pure;
+    #include <math.h>
+    if (x <= 0) return 0;
+    if (df <=0) return 0;
+    double chi = sqrt(x);
+    double term_sum = (df%2==0) ? 1.0  : 0.0;
+    double term = (df%2==0) ? 1.0  : 1.0/chi;
+    double start = (df%2==0) ? 2.0  : 1.0;
+    double stop = (df%2==0) ? (double) (df-2)  : (double) (df-1);
+    for (double ai=start; ai<=stop; ai+=2.0) {
+      term *= x/ai;
+      term_sum += term;
+    }
+    return term_sum;
+  ENDC++;
+  /**
+   * The cumulative distribution function for the Chi Square
+   * distribution.
+   * the CDF for the specfied degrees of freedom.
+   * Translated from the NIST SEL DATAPAC Fortran subroutine CHSCDF.
+   */
+  EXPORT REAL8 Chi2_CDF(REAL8 x, REAL8 df) := FUNCTION
+    INTEGER4 DF_Cut := 1000;
+    REAL8 PI := Core_Const.Pi; //3.14159265358979;
+    INTEGER4 ndf := (INTEGER4) df;
+    BOOLEAN isEven := ndf%2 = 0;
+    REAL8 sd := SQRT(2*df);
+    REAL8 z := (x-df)/sd;
+    // definitions for cases above the cut line for degrees of freedom
+    REAL8 DPWR := 0.33333333333333;
+    REAL8 dfact := 4.5 * df;
+    REAL8 adj_point_gt_x := (POWER(x/df,DPWR)-1.0+1.0/dfact)*SQRT(dfact);
+    REAL8 high_df_gt_x := Normal_CDF(adj_point_gt_x);
+    REAL8 dw := SQRT(x - df-df*LN(x/df));
+    REAL8 root_rdf1 := SQRT(2.0/df);
+    REAL8 root_rdf2 := root_rdf1 * root_rdf1;
+    REAL8 root_rdf3 := root_rdf2 * root_rdf1;
+    REAL8 root_rdf4 := root_rdf3 * root_rdf1;
+    REAL8 d1 := dw;
+    REAL8 d2 := d1 * dw;
+    REAL8 d3 := d2 * dw;
+    REAL8 B11 :=  0.33333333333333;
```

Comment:
I don't know.  I would need to actually do the math to determine the approximating polynomial instead of just using it as defined in the referenced work.